artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/ServerPacketDecoder.java

```diff
@@ -83,16 +85,34 @@
 
 public class ServerPacketDecoder extends ClientPacketDecoder {
 
+   private static final int UUID_LENGTH = 36;
+   private static final int DEFAULT_INTERNER_CAPACITY = 32;
    private static final long serialVersionUID = 3348673114388400766L;
-   public static final ServerPacketDecoder INSTANCE = new ServerPacketDecoder();
+   private SimpleString.Interner keysInterner;
```

Comment:
Surely this is beneficial for clients also, i would look to put the Object pools/caches, down at PacketDecoder level, and have a number of these settable (address, typed property key, typed properly value) .

Reply:
And then configure the consumer also to use it